icaruscode/CRT/CRTCalibrationAnalysis_module.cc

```diff
@@ -0,0 +1,338 @@
+    if(fin.good()) std::cout << "opened file 'feb_map.txt' for reading..." << std::endl;
+    else //std::cout << "could not open file 'feb_map.txt' for reading!" << std::endl;
+        throw cet::exception("CRTDataAnalysis::FillFebMap") << "Unable to find/open file 'feb_map.txt'" << std::endl;
+    std::vector<std::string> row;
+    std::string line, word;
+    while(getline(fin,line)) {
+        row.clear();
+        std::stringstream s(line);
+        int mod;
+        while (std::getline(s, word, ',')) {
+            row.push_back(word);
+        }
+        mod = std::stoi(row[0]);
+        (this->fFebMap)[mod].push_back(std::make_pair(std::stoi(row[1]),std::stoi(row[2])));
+        if(row.size()>3)
+            (this->fFebMap)[mod].push_back(std::make_pair(std::stoi(row[3]),std::stoi(row[4])));
+    }
+    std::cout << "filled febMap with " << (this->fFebMap).size() << " entries" << std::endl;
+    fin.close();
+  }
+
+  
+  //-----------------------------------------------------------------------
+  void CRTCalibrationAnalysis::beginJob()
+  {
+    std::cout << " starting analysis job" << std::endl;
+
+}
+   
+  void CRTCalibrationAnalysis::beginRun(const art::Run& /*run*/)
+  {
+  }
+
+  //-----------------------------------------------------------------------
+  void CRTCalibrationAnalysis::analyze(const art::Event& event) 
+  {
+    MF_LOG_DEBUG("CRT") << "beginning analyis" << '\n';
+
+    // Start by fetching some basic event information for our n-tuple.
+    fEvent  = event.id().event(); 
+    fRun    = event.run();
+    fSubRun = event.subRun();
+
+	std::cout << "event: " << fEvent << ", run " << fRun << ", subrun " << fSubRun << " found." << std::endl;
+
+    FillFebMap();//febMap);
+
+    art::Handle<vector<icarus::crt::CRTData>> crtDAQHandle;
+    bool isCRTDAQ = event.getByLabel(fCRTDAQProducerLabel, crtDAQHandle);
+
+    if (isCRTDAQ)  {
+     std::cout << "about to loop over CRTDAQ entries" << std::endl;
+     for ( auto const& febdat : (*crtDAQHandle) ) {
+        fDetEvent       = fEvent;
+        fMac5           = febdat.fMac5;
+	if(fMac5>93) continue;
+	std::cout << "found mac5: " << fMac5 << std::endl;
+        fEntry          = febdat.fEntry;
+	fFEBReg         = fCrtutils->AuxDetRegionNameToNum(fCrtutils->MacToRegion(fMac5));
+        fNChan = 0;
+	fDetSubSys = fCrtutils->MacToTypeCode(fMac5);
+        fT0 = febdat.fTs0;
+        fT1 = febdat.fTs1;
+
+        int maxchan =0;
+        if(fDetSubSys!=2) maxchan=32;
+        else maxchan = 64;
+	std::cout << "maxchan = " << maxchan << std::endl;
+        for(int ch=0; ch<maxchan; ch++) {
+		std::cout << "channel " << ch << " found!" << std::endl;
+                fADC[ch] = febdat.fAdc[ch]; 
+		macToHistos[fMac5]->at(ch)->Fill( fADC[ch] );
+        } 
+
+//        fDAQNtuple->Fill();
+
+     } //for CRT FEB events
+
+    }//if crtdetsim products present
+
+    else 
+	throw cet::exception("CRTSimAnalysis") << "CRTDAQ products not found!" << std::endl;
```

Comment:
It's going to create confusion if the error message reports the wrong source instead of CRTCalibrationAnalysis. Please see latest commit from @aheggest on CRTDataAnalysis module.